arrow/src/lib.rs

```diff
@@ -21,12 +21,32 @@
 //! Please see the [arrow crates.io](https://crates.io/crates/arrow)
 //! page for feature flags and tips to improve performance.
 //!
+//! # Crate Topology
+//!
+//! The `arrow` project is implemented as multiple sub-crates, which are then re-exported by
+//! this top-level `arrow` crate.
+//!
+//! Crate authors can choose to depend on this top-level `arrow` crate, or just
+//! the sub-crates they need.
+//!
+//! The current list of sub-crates is:
+//!
+//! * [`arrow_buffer`] - buffer abstractions for arrow arrays
+//! * [`arrow_schema`] - the logical types for arrow arrays
+//! * [`arrow_data`] - the underlying data of arrow arrays
+//! * [`arrow_array`] - type-safe arrow array abstractions
+//! * [`arrow_select`] - selection kernels for arrow arrays
```

Comment:
I had to alphabetize them to compare with https://github.com/apache/arrow-rs -- so figured it might help others

```suggestion
//! * [`arrow_array`] - type-safe arrow array abstractions
//! * [`arrow_buffer`] - buffer abstractions for arrow arrays
//! * [`arrow_data`] - the underlying data of arrow arrays
//! * [`arrow_schema`] - the logical types for arrow arrays
//! * [`arrow_select`] - selection kernels for arrow arrays
```

Reply:
Would it also help to mention that some crates, such as `arrow-flight` are meant to be used by themselves (not as an implementation detail of `arrow`)?